controllers/passport.js

```diff
@@ -26,3 +27,40 @@ passport.use(
     }
   )
 );
+
+/**
+ * @FacebookStrategy : The Facebook authentication strategy authenticates users using a Facebook
+ *                     account and OAuth 2.0 tokens.
+ */
+
+passport.use(
+  'facebook',
+  new FacebookStrategy(
+    {
+      clientID: process.env.FACEBOOK_APP_ID,
+      clientSecret: process.env.FACEBOOK_APP_SECRET
+    },
+    async (accessToken, refreshToken, profile, cb) => {
+      try {
+        const user = await User.findOne({ 'facebook.id': profile.id });
+        if (user) {
+          return cb(null, user);
+        }
+        const newUser = new User({
+          method: 'facebook',
+          facebook: {
+            id: profile.id,
+            email: profile.emails[0].value,
+            full_name: profile.displayName,
```

Comment:
please change `full_name`, `first_name`, `last_name` to camel case, every variable or property name should be camel cased.